src/components/Square.js

```diff
@@ -8,12 +8,22 @@ const Square = (props) => {
   //  Component to alert a parent 
   //  component when it's clicked on.
 
+  const onSquareClick = () => {
+    // call the function passed from `App` if the square value is ''
+    // do nothing if there is already some value or the winner is found
+    if (props.value === '' && !props.winner) {
```

Comment:
It's great that you have these two checks! This is somewhat a matter of opinion, I think...but I'd feel inclined to put these checks in the callback function, so that all the "logic" for what happens when you click a square lives in one place.

Reply:
This mostly makes it easier for people to feel confident they understand your program quickly without worrying that there's hidden code they didn't see.